### PR TITLE
Update PyBEL to v0.14

### LINKED ISSUE #993: Update PyBEL to v0.14
I haven't released PyBEL v0.14 yet, but there are a few API changes. I've made a branch [cthoyt/indra@update-pybel-14](https://github.com/cthoyt/indra/tree/update-pybel-14) and the appropriate updates, but I want to make sure everything's working first. There are a couple to-do's:

- [ ] Pin PyBEL to < 0.14.0 in requirements on sorgerlab/indra@master (@bgyori)
- [x] Update https://github.com/cthoyt/bel-selventa to have PyBEL v0.14 cached JSON (@cthoyt)
- [ ] Update AWS instance used in INDRA testing to have PyBEL v0.14 JSON (@bgyori)
- [x] Submit PR from cthoyt/indra@update-pybel-14 to sorgerlab/indra@master (@cthoyt)

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #993 

PyBEL v0.14 has a new data model that better encapsulates named entities, so the nodelink JSON schema has changed a bit. It also has some high-level API changes:

- all `from_*_file` and `from_*_path` functions have been merged using magical decorators)
- renames to reduce some confusion (e.g., the `pybel.from_path` function has been renamed to `pybel.from_belscript_file`, the `from_json_*` series has been renamed to `from_nodelink_*`)
- deprecation of `pybel.resources` from v0.13 has gone into effect, so all imports are updated to `bel_resources`

### CODE REVIEW

**Comment by bgyori** [comment]
I rebased this branch on top of the latest master to follow the preferred git workflow, you can locally reset to the new remote branch to continue. Thanks!

**Comment by cthoyt** [comment]
Hey @bgyori everything seems to be running smoothly with the updates except one test that relies on https://s3.amazonaws.com/bigmech/travis/Hox-2.0-Hs_nljson.json. This file follows the old format - how was it generated? Maybe I can make a new version, otherwise I'll replace the test that relies on this with tests using example BEL graphs already included in PyBEL or we can pick another resource

**Comment by bgyori** [comment]
Thanks! We'll look into replacing the cached test file on S3. The idea would be to wait with this branch until PyBEL 0.14 is released, then revert to simply installing `pybel` in setup.py, and merge, right? Do you have a plan for when you want to do the release?

**Comment by bgyori** [comment]
@cthoyt since you earlier found that things were working, I'm not sure what happened, but when I tried to pip install pybel directly (the current latest version, 0.14.1) a large number of tests failed or errored on Travis, see: https://travis-ci.org/sorgerlab/indra/builds/612455583. Is there a simple solution to this or should we go and fix issues one by one?

**Comment by bgyori** [comment]
Looks like #981 may be behind the test failures that we didn't have on this branch previously, so probably the code added/changed in that PR will need to be adapted to PyBEL 0.14.

**Comment by dianakolusheva** [comment]
Looks like there are two tests that are still failing. First is the one @cthoyt mentioned above where a file needs to be replaced on S3 and another (skipped in PullRequest Travis builds, but you can see it on your personal Travis accounts) is test_pybel_neighborhood_query. For the later, it looks like the file https://github.com/cthoyt/selventa-knowledge/raw/prepare-pybel-14/selventa_knowledge/small_corpus.bel.nodelink.json.gz is also following the old format.

**Comment by bgyori** [comment]
There's already a PR for the Selventa corpus: https://github.com/cthoyt/selventa-knowledge/pull/6
The other one is just a test file that we can try to regenerate or replace with another test network.

**Comment by bgyori** [comment]
Hey @cthoyt, it looks like https://github.com/cthoyt/selventa-knowledge/pull/6 is not yet merged (is that merge it being blocked by another dependency?), which we should wait for before merging this PR. Is there anything else we should look out for before merging?

**Comment by cthoyt** [comment]
I need to update the `bel-repository` repository on PyPI then https://github.com/cthoyt/selventa-knowledge/pull/6 should be ready. Besides that, I think we're ready to go

File: indra/assemblers/pybel/assembler.py [hunk 1]
```diff
@@ -377,34 +385,37 @@ def _assemble_transphosphorylation(self, stmt):
...
-        #cyto_go = cyto_uri.rsplit('/')[-1]
+        # cc = hierarchies['cellular_component']
+        # nuc_uri = cc.find_entity('nucleus')
+        # cyto_uri = cc.find_entity('cytoplasm')
+        # cyto_go = cyto_uri.rsplit('/')[-1]
         pass
 
 
 def belgraph_to_signed_graph(belgraph, symmetric_variant_links=False):
-        edge_set = set()
```
Comment (cthoyt) [inline indra/assemblers/pybel/assembler.py]: not sure why the indentation was wrong here

File: indra/assemblers/pybel/assembler.py [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
-from builtins import dict, str
-import uuid
+
 import logging
+import uuid
+from copy import copy, deepcopy
+
 import networkx as nx
-from copy import deepcopy, copy
+from builtins import dict, str
```
Comment (bgyori) [inline indra/assemblers/pybel/assembler.py]: Why move this stuff around?

File: indra/assemblers/pybel/assembler.py [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
+
 import pybel
 import pybel.constants as pc
+from indra.databases import hgnc_client
+from indra.statements import *
```
Comment (bgyori) [inline indra/assemblers/pybel/assembler.py]: Again, why move these? Typically indra imports should be the last in the order of things according to the standard.

**Comment by bgyori** [comment]
Let's please use rebase/reset and no merges when working on these PRs as explained in the contribution guidelines. It is much easier to review changes that way. I'll try to linearize things and force push.

**Comment by bgyori** [comment]
I have given up on attempting to rebase this due to completely unnecessary style changes that now conflict with master in anoying ways. I'l discuss with @dianakolusheva on how to proceed.

File: indra/assemblers/pybel/assembler.py (hunk 2 above)
Comment (cthoyt) [reply indra/assemblers/pybel/assembler.py]: I used the reformat code function in pycharm and one of the things it takes care of is to sort the imports. I accidentally had PyBEL imported before so it didn't get the order quite right (it assumes if you have other projects open then they should be in the last group of project-related imports)

I've fixed that

**Comment by cthoyt** [comment]
@bgyori sorry about the code style updates, but there's a bit of method to to the madness. Inside the attached zip is `.flake8` and `tox.ini` that run the `flake8` tool with standard settings to check for stuff like proper documentation style, line lengths, correct indentation, etc. 

I've set it up and applied it just to the few pybel-related modules

[Code Style Stuff.zip](https://github.com/sorgerlab/indra/files/3896168/Code.Style.Stuff.zip)

**Comment by bgyori** [comment]
Yes, let's split the PEP8 fixes into a separate PR after this one is merged. Thanks!

**Comment by dianakolusheva** [comment]
I rebased on latest master and resolved conflicts to the best of my knowledge trying to preserve all new pybel functionality and Indra style. Looks like some tests are still failing.

**Comment by cthoyt** [comment]
@bgyori @dianakolusheva the only test that isn't passing is related to processing the node-link JSON file from PyBEL. I created a new version of the Hox network from the CBN that was being used before and force added it to the repo (against the wishes of the gitignore) so you'd have a copy. Maybe you want to stick it on AWS the same way as the old file and revert this test to downloading it then cleaning it up during the tests.

**Pull request merged**